bluecellulab/tools.py

```diff
@@ -26,28 +26,35 @@
 import sys
 from typing import Any, Optional
 import warnings
+import logging
 
 import numpy as np
 
 import bluecellulab
 from bluecellulab import neuron
 
+logger = logging.getLogger(__name__)
 
 VERBOSE_LEVEL = 0
 ENV_VERBOSE_LEVEL: Optional[str] = None
 
 
-def set_verbose(level: int = 1) -> None:
+def set_verbose(level: int = 20) -> None:
     """Set the verbose level of BluecellulabError.
 
     Parameters
     ----------
     level :
             Verbose level, the higher the more verbosity.
-            Level 0 means 'completely quiet', except if some very serious
-            errors or warnings are encountered.
+            # NOTSET (0): allows all messages to be processed.
+            # DEBUG (10)
+            # INFO (20)
+            # WARNING (30)
+            # ERROR (40)
+            # CRITICAL (50)
     """
     bluecellulab.VERBOSE_LEVEL = level
+    logger.setLevel(level)
```

Comment:
Just to be sure. So if a user now call set_verbose(), the user won't have to do 'anything' else right to get the messages (i.e. no import of logging in his own scripts etc).

Reply:
Correct, if a user calls set_verbose() it will configure the logging with a specific level. The user won't need to import the logging module or perform any additional setup in their own scripts to see the log messages.

Reply:
Actually it doesn't work on this case.

```python
Python 3.11.0rc1 (main, Aug 12 2022, 10:02:14) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from bluecellulab import RNGSettings
Setting rng mode to: Random123  # this gets printed as expected
>>> from bluecellulab.tools import set_verbose
>>> set_verbose(50)
>>> rng = RNGSettings()
Setting rng mode to: Random123  # this was not supposed to be printed
```

To solve it we probably need to set the level of root logger to affect all loggers at once.

```suggestion
    logging.getLogger().setLevel(level)  # set level of root logger
```

Reply:
I just noticed the docstring of that method says

> Verbose level, the higher the more verbosity.

Is it not the lower the more verbosity instead?

Reply:
I thought 0 == nothing, 1 a bit, 100 all, etc?

Reply:
They have it the other way around. The higher, the less verbose.

https://docs.python.org/3/library/logging.html#logging-levels

Reply:
I didn't know that the logger names were defining a hierarchy by using dot-notation.
https://stackoverflow.com/a/50755200/1935611

In practice, our logger names are as follows

```python
(Pdb) __name__
'bluecellulab.cell.stimuli_generator'
```

To affect all of the bluecellulab loggers and (only the bluecellulab loggers alone) the following seems to be the solution. 

```python
logging.getLogger('bluecellulab').setLevel(level)
```

Having a unit test for ensuring it would be great.

Reply:
ok, sure, sorry, you were talking about the logging module itself.

Reply:
and yes, the hierarchy might be a solution indeed.

Reply:
> Yeah, but aren't we messing up the user settings with that? 

Yes right. We shouldn't touch the root logger. We should only touch the `bluecellulab.` and its children loggers.

Reply:
I'll make the necessary changes then and also explain more clearly in the comments that the higher the level, the less verbose

Reply:
We should not change the meaning of the argument of set_verbose API though. It should act in a similar way as before.

Reply:
We should keep complete backward compatibility. If in the past somebody set set_verbose to something, the software should act as close as possible to the past behavior

Reply:
Obviously, if the user interacts with the logging module directly, the behavior of the logging module itself doesn't (shouldn't) have to change.
But scripts written in the past (using bglibpy set_verbose e.g.) should still act the same as they did before.